### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by zalmane**

### CODE REVIEW

File: core/tests/test_jmespath_expression.py [hunk 1]
```diff
@@ -215,3 +215,8 @@ def test_jmespath_in():
 
     assert expression.search({"el": 1})
     assert not expression.search({"el": 0})
+
+def test_jmespath_nested_expression():
+    expression.compile(f"a.b")
```
Comment (spicy-sauce) [inline core/tests/test_jmespath_expression.py]: "a.b" is already a string, why f-string is needed?

File: integration-tests/test_stdin_to_stdout.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import json
 import os
 
+import pytest
+
 from common.utils import run_job
```
Comment (spicy-sauce) [inline integration-tests/test_stdin_to_stdout.py]: you should install the isort extension, so the imports will be sorted the same:
```
import json
import os

import pytest
from common.utils import run_job
```

File: integration-tests/test_stdin_to_stdout.py [hunk 3]
```diff
@@ -21,3 +23,25 @@ def test_stdin_to_stdout(tmpdir):
...
     os.remove(output_file)
+
+@pytest.mark.xfail(run=False)
+def test_stdin_to_stdout_filtered(tmpdir):
+    tested_data = '{"id": 121, "fname": "joe", "lname": "allen", "country_code": "US", "country_name": "united states", "gender": "M"}'
+    tested_data2 = '{"id": 122, "fname": "shmoe", "lname": "allen", "country_code": "US", "country_name": "united states", "gender": "M"}'
+    tested_data3 = '{"id": 123, "fname": "joe", "lname": "allen", "country_code": "US", "country_name": "united states", "gender": "M"}'
+
+    output_file = tmpdir.join("test_stdin_to_stdout.txt")
+    run_job("tests.stdin_to_stdout", f"echo '{tested_data}\n{tested_data2}\n{tested_data3}'", output_file)
```
Comment (spicy-sauce) [inline integration-tests/test_stdin_to_stdout.py:32]: why f-string is needed? it's already a string...

File: integration-tests/test_stdin_to_stdout.py [hunk 4]
```diff
@@ -21,3 +23,25 @@ def test_stdin_to_stdout(tmpdir):
     assert result.get("gender") == "M"
 
     os.remove(output_file)
+
+@pytest.mark.xfail(run=False)
```
Comment (spicy-sauce) [inline integration-tests/test_stdin_to_stdout.py]: why is it marked with xfail?

**Review by spicy-sauce — CHANGES_REQUESTED** [review]

File: core/tests/test_jmespath_expression.py (hunk 1 above)
Comment (zalmane) [reply core/tests/test_jmespath_expression.py]: fixed

File: integration-tests/test_stdin_to_stdout.py (hunk 3 above)
Comment (zalmane) [reply integration-tests/test_stdin_to_stdout.py:32]: fixed

File: integration-tests/test_stdin_to_stdout.py (hunk 2 above)
Comment (zalmane) [reply integration-tests/test_stdin_to_stdout.py]: have some conflict with this plugin. aren't they sorted now?

File: integration-tests/test_stdin_to_stdout.py (hunk 3 above)
Comment (zalmane) [reply integration-tests/test_stdin_to_stdout.py:32]: it has interpolation of tested data

File: integration-tests/test_stdin_to_stdout.py (hunk 4 above)
Comment (zalmane) [reply integration-tests/test_stdin_to_stdout.py]: removed

Comment (zalmane) [reply integration-tests/test_stdin_to_stdout.py]: it is related to the filter bug. will be resolved with that PR.

File: integration-tests/test_stdin_to_stdout.py (hunk 2 above)
Comment (spicy-sauce) [reply integration-tests/test_stdin_to_stdout.py]: no, and `pytest` is unused. I'm fixing.

File: integration-tests/test_stdin_to_stdout.py (hunk 3 above)
Comment (spicy-sauce) [reply integration-tests/test_stdin_to_stdout.py:32]: yes, my mistake.

**Review by spicy-sauce — APPROVED** [review]

**Pull request merged**